### DISCUSSION TITLE
Automatically allow a cookie if a service is accepted

### DISCUSSION THREAD

**Discussion by bigahuna** [eibiflo/cf_cookiemanager#49] [Ideas]
Hi all, 

First of all, thanks for this great extension.

I'm wondering if it could be made possible that if someone enables i.e. a youtube video, that the option to allow youtube videos is automatically added to the cf_cookie in the same way it would be added if the visitor enables youtube cookies in the cookie banner.

Example:

Before:

```{"categories":["cfcookiemanager"],"level":["cfcookiemanager"], ... }```

After enabling a youtube video (load a single youtube video)

```{"categories":["youtube","cfcookiemanager"],"level":["youtube","cfcookiemanager"], ... }```

If this is somehow already possible and I did not find the optiom, please excuse this post.

Regards
Mike

**Comment by eibiflo** [comment]
Hi @bigahuna 

Thanks for the great Feedback,  feel free to leave a like/star - that always motivates me! 

If I understand correctly, you want to accept the cookies automatically with a click on the iframe manager, similar to the user consent banner.

This isn't directly possible, because of the Code-Design, but you can use the Cookie Manager JavaScript API.
More information here: [Documentation](https://docs.typo3.org/p/codingfreaks/cf-cookiemanager/main/en-us/Developer/JavascriptAPI/Index.html)

**Quick Example**
```
// Select all elements with the data-service="youtube" attribute
const youtubeElements = document.querySelectorAll('[data-service="youtube"]');


// Add a click event listener to each element
youtubeElements.forEach(element => {
    element.addEventListener('click', () => {
        // Call the cc.accept function from the Javascript API  with 'youtube' as the argument
        cc.accept('youtube');
    });
});

```

**Reply by bigahuna** [reply]
Great, thanks for your answer! I will try it out. Regards, Mike

**Comment by bigahuna** [comment]
Works like a charm. Thanks again!
